python/paddle/fluid/tests/unittests/dygraph_to_static/test_param_guard.py

```diff
@@ -91,5 +91,57 @@ def test_parameter_list_iter(self):
                                                                    static_loss))
 
 
+class NetWithRawParamList(paddle.nn.Layer):
+    def __init__(self, in_size, out_size):
+        super(NetWithRawParamList, self).__init__()
+        weight = self.add_parameter('w',
+                                    self.create_parameter([in_size, out_size]))
+        bias = self.add_parameter(
+            'b', self.create_parameter(
+                [out_size], is_bias=True))
+        self.params = [weight]
+        self.bias_dict = {'b': bias}
+
+    @to_static
+    def forward(self, x):
+        out = paddle.matmul(x, self.params[0])
+        out = paddle.add(out, self.bias_dict['b'])
+        out = paddle.tanh(out)
+        return out
+
```

Comment:
Do you know whether we can support the following case? If yes, could we add a testcase to make sure of it?
```
class NewClass(paddle.nn.Layer):
    def __init__(self, old_class): # old_class can be your NetWithRawParamList
        ...
        # The key difference is that params come from outer side of the class
        self.params = [old_class.weight]
        self.bias_dict = {'b': old_class.bias}

    @to_static
    def forward(self, x):
        out = paddle.matmul(x, self.params[0])
        out = paddle.add(out, self.bias_dict['b'])
        out = paddle.tanh(out)
        return out
```

Reply:
Thanks，add more unittest